guides/web-apps/acceptance-testing-with-selenium/README.md

```diff
@@ -114,157 +84,61 @@ We don't want to test only Firefox. Because of that, you should use Selenium Rem
 
-Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.0.0-beta3.jar`.
+- Wait for a text element to be present before you check the content of that element
 
-Now, press "Run All" and relax. This will take a while as the system runs your test in three different browsers!
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs)	
+	- Please take a look at [TextareaPage_WriteToTextArea()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L29) test and [line 37](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L37).	
+	- Method `WaitForText()` is use to wait for text in `TextareaInfoLabel` for maximum 5 second. Then assert is true and test pass.
 
-![test explorer screenshot](/assets/2016-04-01-15_38_44-Launcher-Microsoft-Visual-Studio.png)
+- Wait for a button to be present before you click on that button
 
-## Wait for asynchronous content
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs)	
+	- As you can see in [line 35](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs#L35) in [ButtonPage_RegularButton()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs) test we use one of the helper method called `WaitUntil`.	
+	- As parameter we use `Displayed` property for the button. The test will wait until `ButtonInlineScript` is displayed. Default maximum wait time is 10 second.
 
-There is one common pitfall when writing Selenium tests. The test is executed with disregard of the asynchronously loaded content. This means that your tests need to explicitly wait for UI elements to appear before running any actions or assertions on them.
+- Wait for presence of an input field before typing in it and wait for text to be present in label
 
-It is a good practice to always wait:
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextPageTest.cs)
+	- In [line 31](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextPageTest.cs#L31) in [TextPage_TextPropagationOnUnfocus()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextPageTest.cs) test we use `WaitUntil` helper method again to wait for input to be displayed.	
+	- In [line 34](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextPageTest.cs#L34) `WaitForText()` is used to wait for text in label for maximum 5 second.
```

Comment:
- An **example** can be found in the method `TextPage_TextPropagationOnUnfocus` in the KitchenSink tests (see [TextPageTest.cs lines 28-38](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextPageTest.cs#L28-L38)). This test mixes the other examples presented above. The method `WaitUntil()` is used here to asynchronously wait for the input to be `Displayed`. Notice that the following response check is also done asynchrously using `WaitForText `.